eos/b-decays/b-to-3l-nu_TEST.cc

```diff
@@ -0,0 +1,55 @@
+/* vim: set sw=4 sts=4 et foldmethod=syntax : */
+
+/*
+ * Copyright (c) 2022 Danny van Dyk
+ * Copyright (c) 2022 Stephan Kürten
+ *
+ * This file is part of the EOS project. EOS is free software;
+ * you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License version 2, as published by the Free Software Foundation.
+ *
+ * EOS is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59 Temple
+ * Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <test/test.hh>
+#include <eos/b-decays/b-to-3l-nu.hh>
+
+using namespace test;
+using namespace eos;
+
+class BToThreeLeptonsNeutrinoTest :
+    public TestCase
+{
+    public:
+        BToThreeLeptonsNeutrinoTest() :
+            TestCase("b_to_3l_nu_test")
+        {
+        }
+
+        virtual void run() const
+        {
+
+            static const double eps = 1e-28;
+
+            Parameters p = Parameters::Defaults();
+
+            {
+
+                BToThreeLeptonsNeutrino three_lnu(p, Options{ });
+
+
+                //TEST_CHECK_NEARLY_EQUAL(three_lnu.branching_ratio_5diff(2.0, 2.0, 1.0, 1.0, 0.1), 1.0, eps);
+                TEST_CHECK_NEARLY_EQUAL(three_lnu.decay_width_2diff(2.0, 2.0), 3.3877771208595474e-24, 0.5e-24);
+                TEST_CHECK_NEARLY_EQUAL(three_lnu.branching_ratio_2diff(2.0, 2.0), 8.430686902290666e-12, 0.5e-12);
+                TEST_CHECK_NEARLY_EQUAL(three_lnu.branching_ratio_2diff(0.6, 0.5), 7.688928783225041e-09, 0.5e-09);
+                //TEST_CHECK_NEARLY_EQUAL(three_lnu.branching_ratio_2diff(2.0, 2.0), 5, eps);
```

Comment:
Please use a more systematic grid for the kinematic variables, e.g.,
 - 2 different values of q2;
 - 2 different values of k2;
 - 3 different values of z_gamma;
 - 3 different values of z_W;
 - 3 different values of phi;